src/entities/trades/sushiswap/Sushiswap.ts

```diff
@@ -0,0 +1,291 @@
+        //     "swapPrice": 9.99757e-13,
+        //     "priceImpact": 0.0005750648583481288,
+        //     "amountIn": "1000000000000000000",
+        //     "assumedAmountOut": "999757",
+        //     "gasSpent": 110000,
+        //     "route": [
+        //         {
+        //             "poolAddress": "0xF5e270c0D97F88EFB023a161b9Fcc5d0C7ad0b70",
+        //             "poolType": "Concentrated",
+        //             "poolName": "SushiSwapV3 0.01%",
+        //             "poolFee": 0.0001,
+        //             "tokenFrom": 0,
+        //             "tokenTo": 1,
+        //             "share": 1,
+        //             "assumedAmountIn": "1000000000000000000",
+        //             "assumedAmountOut": "999757"
+        //         }
+        //     ],
+        //     "routeProcessorAddr": "0x7A4af156379f512DE147ed3b96393047226d923F",
+        //     "routeProcessorArgs": {
+        //         "tokenIn": "0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d",
+        //         "amountIn": "1000000000000000000",
+        //         "tokenOut": "0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83",
+        //         "amountOutMin": "994758",
+        //         "to": "0x2B1a6dD2a80f7e9A2305205572Df0F4B38b205A1",
+        //         "routeCode": "0x02e91D153E0b41518A2Ce8Dd3D7944Fa863463a97d01ffff01F5e270c0D97F88EFB023a161b9Fcc5d0C7ad0b70002B1a6dD2a80f7e9A2305205572Df0F4B38b205A1"
+        //     }
+        // }
+
+        return new SushiswapTrade({
+          maximumSlippage,
+          inputAmount: currencyAmountIn,
+          outputAmount: currencyAmountOut,
+          tradeType,
+          chainId,
+          approveAddress,
+          priceImpact: new Percent('0', '1'),
+          routeCode,
+        })
+      }
+    } catch (error) {
+      console.error('OneInch.getQuote: Error fetching the quote:', error.message)
+      return null
+    }
+
+    return null
+  }
+
+  public minimumAmountOut(): CurrencyAmount {
+    if (this.tradeType === TradeType.EXACT_OUTPUT) {
+      return this.outputAmount
+    } else {
+      const slippageAdjustedAmountOut = new Fraction(ONE)
+        .add(this.maximumSlippage)
+        .invert()
+        .multiply(this.outputAmount.raw).quotient
+      return this.outputAmount instanceof TokenAmount
+        ? new TokenAmount(this.outputAmount.token, slippageAdjustedAmountOut)
+        : CurrencyAmount.nativeCurrency(slippageAdjustedAmountOut, this.chainId)
+    }
+  }
+
+  public maximumAmountIn(): CurrencyAmount {
+    if (this.tradeType === TradeType.EXACT_INPUT) {
+      return this.inputAmount
+    } else {
+      const slippageAdjustedAmountIn = new Fraction(ONE)
+        .add(this.maximumSlippage)
+        .multiply(this.inputAmount.raw).quotient
+      return this.inputAmount instanceof TokenAmount
+        ? new TokenAmount(this.inputAmount.token, slippageAdjustedAmountIn)
+        : CurrencyAmount.nativeCurrency(slippageAdjustedAmountIn, this.chainId)
+    }
+  }

```

Comment:
not directly related: 

these methods seems to be always the same, maybe we could move them to base class and inherit them.

Reply:
There are places where this code is different, but agree that we should find a solution for this.